File: java/example/src/global/hiber/example/Authenticated.java
package global.hiber.example;

import io.grpc.*;
import java.util.concurrent.Executor;

class Authenticated implements CallCredentials {

  private final String token;

  Authenticated(String token) {
    this.token = token;
  }

  @Override
  public void thisUsesUnstableApi() {
  }

  @Override
  public void applyRequestMetadata(
    MethodDescriptor<?, ?> method, Attributes attrs,
    Executor appExecutor, MetadataApplier applier) {

    Metadata metadata = new Metadata();
    metadata.put(Metadata.Key.of("authorization", Metadata.ASCII_STRING_MARSHALLER), "Bearer " + token);

    applier.apply(metadata);
  }
}

File: kotlin/example/src/global/hiber/example/Authenticated.kt
package global.hiber.example;

import io.grpc.*
import java.util.concurrent.Executor

class Authenticated(val token: String) : CallCredentials {
  override fun thisUsesUnstableApi() = Unit
  override fun applyRequestMetadata(
    method: MethodDescriptor<*, *>, attrs: Attributes,
    appExecutor: Executor, applier: CallCredentials.MetadataApplier
  ) = applier.apply(Metadata().withToken(token))

  private fun Metadata.withToken(token: String): Metadata = apply { put(authorizationKey, "Bearer $token") }

  companion object {
    private val authorizationKey = Metadata.Key.of("authorization", Metadata.ASCII_STRING_MARSHALLER)
  }
}
